app/models/remote_authenticatable.rb

```diff
@@ -0,0 +1,119 @@
+module Devise
+  module Models
+    module RemoteAuthenticatable
+      extend ActiveSupport::Concern
+
+      #
+      # Here you do the request to the external webservice
+      #
+      # If the authentication is successful you should return
+      # a resource instance
+      #
+      # If the authentication fails you should return false
+      #
+      #def remote_authentication(authentication_hash)
+      def remote_authentication(params)
+        if Rails.application.secrets.cognito_aws_access_key_id.present? &&
```

Comment:
Rails.application.secrets.cognito_aws_access_key_id and similar terms are repeated alot here. Could they be moved into helper methods that are a bit shorter?

Reply:
Moved this out to helper methods... May want to create a module to cover the whole app as we have 2 lots of aws creds in the app now.